tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +106,136 @@ def GetFieldsForTypeName(
 
     return entity_type_fields_sorted
 
-  def _CalculateMatchWeight(self, concrete_fields: Set[EntityTypeField],
-                            canonical_fields: Set[EntityTypeField]) -> float:
-    """Determines the weight of a match and returns that weight as a float.
+  def _CalculateMatchScore(self, concrete_fields: Set[StandardField],
+                           canonical_fields: Set[EntityTypeField])-> float:
+    """Calculates a match's score.
 
-    Finds the size of the intersection(x) between the two sets of fields and
-    the size of the sets of fields unique to the concrete entity and canonical
-    type(y and z). These three variables are input into the function
-    f(x,y,z) = 1/((x^2)+(z-y)) to determine the weight.
+    The score of a match is determined by calculating the average of two
+    f-scores. The first f-score is the measure of correctly matched required
+    fields in the canonical type and the second f-score is the measure of
+    correctly matched fields regardless of optionality. Adding the two f-scores
+    creates a preference for matches with a higher quantity of matched required
+    canonical fields, and dividing by 2 keeps the range of the function in
+    [-1, 1].
 
     Args:
-      concrete_fields: A set of EntityTypeField objects belonging to the
+      concrete_fields: A set of StandardField objects belonging to the
         concrete entity being matched.
       canonical_fields: A set of EntityTypeField objects belonging to an Entity
         Type defined in DBO.
 
     Returns:
-      The weight of the match as a floating point number.
+      A match's score as a floating point number.
     """
 
-  def _CreateMatch(self, field_list: List[EntityTypeField],
+    required_canonical_fields = {
+        model.StandardizeField(field) for field in canonical_fields
+        if not field.IsOptional()
+    }
+
+    standard_canonical_fields: Set[StandardField] = set()
+    for field in canonical_fields:
+      standard_canonical_fields.add(model.StandardizeField(field))
+
+    ma = len(concrete_fields.intersection(standard_canonical_fields))
+    mr = len(concrete_fields.intersection(required_canonical_fields))
+    tr = len(required_canonical_fields)
+    c = len(concrete_fields)
+    e = len(concrete_fields.difference(standard_canonical_fields))
+    a = len(required_canonical_fields.difference(concrete_fields))
```

Comment:
These variables could use more descriptive names. It might make the actual formula lines clunkier, but I think it would make the function overall easier to understand.

Reply:
Not only does it make the equation clunkier, but now I have to separate everything out into several lines in order to meet the linter's line lenth requirements. The code is now a lot harder to read. This could be solved with comments instead of changing the variable names.